maraboupy/MarabouNetworkComposition.py

```diff
@@ -0,0 +1,233 @@
+'''
+Top contributors (to current version):
+    - Teruhiro Tagomori
+
+This file is part of the Marabou project.
+Copyright (c) 2017-2019 by the authors listed in the file AUTHORS
+in the top-level source directory) and their institutional affiliations.
+All rights reserved. See the file COPYING in the top-level source
+directory for licensing information.
+
+MarabouNetworkComposition represents neural networks with piecewise linear constraints derived from the ONNX format
+'''
+
+import numpy as np
+import onnx
+import sys
+import pathlib
+import os
+sys.path.insert(0, os.path.join(str(pathlib.Path(__file__).parent.absolute()), "../"))
+
+from maraboupy import Marabou
+from maraboupy import MarabouCore
+from maraboupy import MarabouNetwork
+from maraboupy import MarabouNetworkONNX
+
+class MarabouNetworkComposition(MarabouNetwork.MarabouNetwork):
+    """Constructs a MarabouNetworkComposition object from an ONNX file
+
+    Args:
+        filename (str): Path to the ONNX file
+        inputNames: (list of str, optional): List of node names corresponding to inputs
+        outputNames: (list of str, optional): List of node names corresponding to outputs
+        reindexOutputVars: (bool, optional): Reindex the variables so that the output variables are immediate after input variables.
+        threshold: (float, optional): Threshold for equation functions. If the absolute value of the lower bound of a variable is less than this value, the variable is considered to be 0.
+
+    Returns:
+        :class:`~maraboupy.Marabou.MarabouNetworkComposition.MarabouNetworkComposition`
+    """
+    def __init__(self, filename, inputNames=None, outputNames=None, reindexOutputVars=True, threshold=None):
+        super().__init__()
+        self.shapeMap = {}
+        self.madeGraphEquations = []
+        self.ipqs = []
+        self.ipqToInVars = {}
+        self.ipqToOutVars = {}
+        self.inputVars, self.outputVars = self.getInputOutputVars(filename, inputNames, outputNames)
+
+        network = MarabouNetworkONNX.MarabouNetworkONNX(filename, reindexOutputVars=reindexOutputVars, threshold=threshold)
+
+        network.saveQuery('q1.ipq')
```

Comment:
Lots and lots of hard coded names here. At the very least they need to be heavily documented, but it would be better to make them configurable.

Reply:
reindexOutputVars was deleted since it is required